stan/math/rev/mat/fun/softmax.hpp

```diff
@@ -4,81 +4,78 @@
 #include <stan/math/prim/arr/err/check_nonzero_size.hpp>
 #include <stan/math/prim/mat/fun/Eigen.hpp>
 #include <stan/math/prim/mat/fun/softmax.hpp>
-#include <stan/math/rev/core.hpp>
+#include <stan/math/rev/mat.hpp>
 #include <vector>
 
 namespace stan {
 namespace math {
 
 namespace {
-class softmax_elt_vari : public vari {
- private:
-  vari** alpha_;
-  const double* softmax_alpha_;
-  const int size_;  // array sizes
-  const int idx_;   // in in softmax output
-
+class softmax_op {
```

Comment:
There should be an explicit nullary constructor setting the pointer to `null` and `N_` to zero.